src/tls/connection.c

```diff
@@ -719,8 +721,11 @@ verify_peer_certificate (gnutls_session_t session)
 
   if (ret == 0)
     {
-      /* ignore CA/trusted owner and leave that to e. g. sssd */
-      status &= ~(GNUTLS_CERT_INVALID | GNUTLS_CERT_SIGNER_NOT_FOUND | GNUTLS_CERT_SIGNER_NOT_CA);
+      if (!parameters.client_cert_ca)
+        {
+          /* ignore CA/trusted owner without CA */
+          status &= ~(GNUTLS_CERT_INVALID | GNUTLS_CERT_SIGNER_NOT_FOUND | GNUTLS_CERT_SIGNER_NOT_CA);
```

Comment:
I've always kinda disliked this.  Could we rather add an `ignore_mask` variable and set it accordingly?  (making sure to also use it for printing the status).